src/modules/collections/services/collections.service.ts

```diff
@@ -63,42 +69,56 @@ export class CollectionsService {
 	/**
 	 * Adapt a collection as returned by a typical graphQl response to our internal collection data model
 	 */
-	public adaptCollection(gqlCollection: GqlCollection | undefined): Collection | undefined {
+	public async adaptCollection(
+		gqlCollection: GqlCollection | undefined
+	): Promise<Collection | undefined> {
 		if (!gqlCollection) {
 			return undefined;
 		}
+
 		return {
 			id: gqlCollection.id,
 			name: gqlCollection.name,
 			userProfileId: gqlCollection.user_profile_id,
 			createdAt: gqlCollection.created_at,
 			updatedAt: gqlCollection.updated_at,
 			isDefault: gqlCollection.is_default,
-			objects: (gqlCollection as GqlCollectionWithObjects).ies?.map(
-				this.adaptCollectionObjectLink
+			objects: await Promise.all(
+				(gqlCollection as GqlCollectionWithObjects).ies
+					? (gqlCollection as GqlCollectionWithObjects).ies.map((object) =>
+							this.adaptCollectionObjectLink(object)
+					  )
+					: []
 			),
 		};
 	}
 
-	public adaptCollectionObjectLink = (
+	public async adaptCollectionObjectLink(
 		gqlCollectionObjectLink: CollectionObjectLink | undefined
-	): IeObject | undefined => {
+	): Promise<IeObject | undefined> {
 		if (!gqlCollectionObjectLink) {
 			return undefined;
 		}
-		const func = this.adaptIeObject;
-		const objectIe = func(get(gqlCollectionObjectLink, 'ie'));
+		const objectIe = this.adaptIeObject(get(gqlCollectionObjectLink, 'ie'));
+		const resolvedThumbnailUrl = await this.playerTicketService.resolveThumbnailUrl(
+			objectIe.thumbnailUrl,
+			this.referer
+		);
```

Comment:
This will still trigger multiple calls to the ticket service the first time when the token isn't in the cache yet.
eg:
* collection is reaquested
* collection has multiple objects
* each object tries to resolve its thumbnail in parallel
* playerTicketService.getThumbnailToken is called multiple times
* the first one that is called, will start a request to the ticket service
* network requests are slow, so the second time getThumbnailToken is called, the cache is not filled yet
* so the second call also triggers a requests, etc...

You can probably avoid this by using the .wrap() function of the cache-manager package that you are already using:
```
try {
  let token = await memoryCache.wrap(`thumbnailToken-${referer}`, function() {
    return this.getToken(thumbnailPath, referer);
  });
} catch (err) {
  // error handling
}
```

Reply:
fixed